library/big-float/impl-default.lisp

```diff
@@ -675,7 +675,8 @@ returns the nth SeriesSplit, return the series evaluated to the Nth element."
     (define (asin x)
       (atan (* x (reciprocal-sqrt (- 1 (* x x))))))
     (define (acos x)
-      (- (scale (bf-pi) -1) (asin x))))
+      (- (scale (bf-pi) -1) (asin x)))
+    (define pi (BFConst (bf-pi))))
```

Comment:
The purpose of `BFConst` is to capture a _function_ which computes a constant value. We can't `(define pi (bf-pi))` because then some fixed precision of pi will be computed and stored, and it will never react to an increase in precision. Hence we have another kind of "big-float" which is a [`BFConst` which holds on to a function `Unit -> Big-Float`](https://github.com/coalton-lang/coalton/blob/main/library/big-float/impl-default.lisp#L129) which (re-)computes the constant value at the time it's used.